Antidote Verifier/Sdl.Community.AntidoteVerifier/Sdl.Community.AntidoteVerifier/Utils/SegmentExtenstions.cs

```diff
@@ -5,110 +5,162 @@
-                if (text != null)
-                {
-                    var startLocationInsideTextItem = new TextLocation(counter.CurrentLocation, startIndex - counter.CharacterCount);
-                    var segmentText = text.Properties.Text;
-                    return segmentText.Substring(startLocationInsideTextItem.TextOffset,
-                        endPosition - startLocationInsideTextItem.TextOffset);
-
-                }
+                return segmentText.Substring(startIndex, endPosition - startIndex);
             }
 
+            //var counter = segment.GetCharacterCountingIterator(startIndex);
+            //if (counter.CharacterCount < startIndex)
+            //{
+            //	// the actual count is between this location and the next
+            //	//  - if this is a text node we can point to the exact location inside the text
+            //	IText text = counter.CurrentLocation.ItemAtLocation as IText;
+            //	if (text != null)
+            //	{
+            //		var startLocationInsideTextItem = new TextLocation(counter.CurrentLocation, startIndex - counter.CharacterCount);
+            //		var segmentText = text.Properties.Text;
+            //		return segmentText.Substring(startLocationInsideTextItem.TextOffset,
+            //				endPosition - startLocationInsideTextItem.TextOffset);
+
+            //	}
+            //}
+
             return string.Empty;
         }
 
-        public static void Replace(this ISegment segment, int startIndex, int endPosition, string replacementText)
+        //displayLanguage = language of the message or the explication 
+        public static bool CanReplace(this ISegment segment, int startIndex, int endPosition, string origString, string displayLanguage, ref string message, ref string explication)
         {
-            var counter = segment.GetCharacterCountingIterator(startIndex);
-            if (counter.CharacterCount < startIndex)
+            bool ret = false;
+
+            if (segment.Properties.IsLocked)
             {
-                // the actual count is between this location and the next
-                //  - if this is a text node we can point to the exact location inside the text
-                IText text = counter.CurrentLocation.ItemAtLocation as IText;
-                if (text != null)
+                ret = false;
+                if (displayLanguage == "fr")
                 {
-                    var startLocationInsideTextItem = new TextLocation(counter.CurrentLocation, startIndex - counter.CharacterCount);
-                    
-                    var sb = new StringBuilder(text.ToString());
-                    sb.Remove(startLocationInsideTextItem.TextOffset, endPosition - startIndex);
-                    sb.Insert(startLocationInsideTextItem.TextOffset, replacementText);
-                    text.Properties.Text = sb.ToString();
-
+                    message = "Antidote ne peut effectuer la correction, car le segment est en lecture seule.";
```

Comment:
Why does the message has to come from the Studio plugin? Looking at the code it seems to me that this message is displayed when the result of CanReplace is false which means Antidote can decide what message to display rather than coming from the Studio plugin.